org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormatterSettings.java

```diff
@@ -0,0 +1,131 @@
+package org.eclipse.lsp4xml.settings;
+
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+import javax.swing.text.TabableView;
+
+import org.eclipse.lsp4j.FormattingOptions;
+
+/**
+ * This class is the root of all formatting settings.
+ * It is necessary to update this class for any new additions.
+ * 
+ * All defaults should be set here to eventually be overriden if needed.
+ */
+public class XMLFormatterSettings {
```

Comment:
I think it should b ebetter to extends FormattingOptions. It already defines getTabSize and isInsertSpaces (from LSP spec).